extendr-api/src/iter.rs

```diff
@@ -222,13 +225,25 @@ mod tests {
         with_r(|| {
             let single_charsxp = blank_string();
             let s1: Vec<_> = single_charsxp.as_str_iter().unwrap().collect();
-            // dbg!(&s1);
             let single_charsxp = blank_scalar_string();
             let s2: Vec<_> = single_charsxp.as_str_iter().unwrap().collect();
-            // dbg!(&s2);
             assert_eq!(s1, s2);
             assert_eq!(s1.len(), 1);
             assert_eq!(s2.len(), 1);
         });
     }
+
+    #[test]
+    fn test_new_constructor() {
+        with_r(|| {
+            let str_iter = StrIter::new(10);
+            assert_eq!(str_iter.collect::<Vec<_>>().len(), 10);
+            let str_iter = StrIter::new(0);
+            let str_iter_collect = str_iter.collect::<Vec<_>>();
+            assert_eq!(str_iter_collect.len(), 0);
+            assert!(str_iter_collect.is_empty());
```

Comment:
This one confuses me a little bit. Because the check:

> `let vector = if len == 0 { nil_value() } else { na_string() };`

Wouldn't this result in a length 1 NULL list when `len == 0`?

Reply:
I don't think. I'm going to answer with a test (on this PR branch of course):

```rs
#[test]
fn iterating_unamed_list() {
    with_r(|| {
        let unamed_list = List::from_values([1, 2, 3, 42, 100]);
        assert_eq!(unamed_list.iter().len(), 5);
        let unamed_list = List::from_values::<[i32; 0]>([]);
        assert_eq!(unamed_list.iter().len(), 0);

        dbg!(unamed_list.iter().collect::<Vec<_>>());
    });
}
```

---- wrapper::list::tests::iterating_unamed_list stdout ----
[extendr-api/src/wrapper/list.rs:450:13] unamed_list.iter().collect::<Vec<_>>() = []
```

`StrIter` is only used with `.iter()`. What do you think? Resolved?